### PR TITLE
RELATED: RAIL-2760 Move Dropdown and related components from goodstrap to sdk-ui-kit

### PR DESCRIPTION
**Pull request by kandl**
- Move Dropdown and related components from goodstrap to gooddata-ui-sdk
- Minor fixes, typing fixes
- Add missing tests
- Add ResponsiveContext to allow breakpoints customization

JIRA: RAIL-2760, subtask: RAIL-2850

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by kandl** [comment]
extended test

File: libs/sdk-ui-kit/src/Dropdown/DropdownButton.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    isOpen,
+    isSmall = true,
+    iconLeft,
+
+    onClick,
+}) => {
+    const buttonClasses = cx(
+        {
+            "gd-button-primary": true,
```
Comment (no23reason) [inline libs/sdk-ui-kit/src/Dropdown/DropdownButton.tsx:41]: Nitpick: can we move the `: true` classes outside of the object?

File: libs/sdk-ui-kit/src/Dropdown/DropdownList.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        renderNoData = defaultNoData,
+        ...listProps
+    } = props;
+
+    const hasNoData = !isLoading && itemsCount === 0;
+    const hasNoMatchingData = hasNoData && !!searchString;
+
+    const listClassNames = cx("gd-infinite-list", className);
+
+    const searchFieldClassNames = cx({
```
Comment (no23reason) [inline libs/sdk-ui-kit/src/Dropdown/DropdownList.tsx:104]: Nitpick: can we move the `: true` classes outside of the object?

File: libs/sdk-ui-kit/src/Dropdown/tests/customDictionary.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// (C) 2020 GoodData Corporation
+
+// TODO remove this adhoc translations when NoData component will have own dictionary and dont rely on provided Intl from top
```
Comment (no23reason) [inline libs/sdk-ui-kit/src/Dropdown/tests/customDictionary.ts:3]: What does this mean?

File: libs/sdk-ui-kit/src/NoData/NoData.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ */
+export const NoData: React.FC<INoDataProps> = ({
+    className,
+    hasNoMatchingData,
+    notFoundLabel,
+    noDataLabel,
+}) => {
+    const classNames = cx(
+        {
+            "gd-no-data": true,
```
Comment (no23reason) [inline libs/sdk-ui-kit/src/NoData/NoData.tsx:26]: Nitpick: can we move the `: true` classes outside of the object?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1977/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3608/ : SUCCESS in 11m 11s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1978/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/578/ : SUCCESS in 26m 23s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/636/ : FAILURE in 3m 04s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/897/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3610/ : SUCCESS in 21m 11s

**Pull request merged**